core/src/test/java/com/redhat/lightblue/client/request/TestDataUpdateRequest.java

```diff
@@ -46,4 +47,31 @@ public void testProjectionsAsArray() {
                 "\"projection\":{\"field\":\"*\",\"include\":true,\"recursive\":false}"));
     }
 
+    @Test
+    public void testExecutionReadPreference() {
+        DataUpdateRequest request = new DataUpdateRequest("fake");
+        request.execution(Execution.MongoController.withReadPreference(Execution.MongoController.ReadPreference.primaryPreferred));
```

Comment:
@paterczm - I know this PR has already been merged, but is there a way we could short cut this call? Is MongoController even needed?